Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc

```diff
@@ -0,0 +1,157 @@
+// -*- C++ -*-
+/*
+//\class RPCNumberingScheme
+
+ Description: RPC Numbering Scheme for DD4hep
+              
+//
+// Author:  Sergio Lo Meo (sergio.lo.meo@cern.ch) following what Ianna Osburne made for DTs (DD4HEP migration)
+//          Created:  Fri, 20 Sep 2019 
+*/
+#include "Geometry/MuonNumbering/interface/DD4hep_RPCNumberingScheme.h"
+#include "Geometry/MuonNumbering/interface/DD4hep_MuonNumbering.h"
+#include "DataFormats/MuonDetId/interface/RPCDetId.h"
+#include "Geometry/MuonNumbering/interface/MuonBaseNumber.h"
+
+#include <cassert>
+
+using namespace cms;
+
+RPCNumberingScheme::RPCNumberingScheme(const MuonConstants& muonConstants) { initMe(muonConstants); }
+
+void RPCNumberingScheme::initMe(const MuonConstants& muonConstants) {
+  int levelPart = get("level", muonConstants);
+
+  assert(levelPart != 0);
+  theRegionLevel = get("mr_region", muonConstants) / levelPart;
+  theBWheelLevel = get("mr_bwheel", muonConstants) / levelPart;
+  theBStationLevel = get("mr_bstation", muonConstants) / levelPart;
+  theBPlaneLevel = get("mr_bplane", muonConstants) / levelPart;
+  theBChamberLevel = get("mr_bchamber", muonConstants) / levelPart;
+  theEPlaneLevel = get("mr_eplane", muonConstants) / levelPart;
+  theESectorLevel = get("mr_esector", muonConstants) / levelPart;
+  theERollLevel = get("mr_eroll", muonConstants) / levelPart;
+}
+
+void RPCNumberingScheme::baseNumberToUnitNumber(const MuonBaseNumber& num) {
+  const int barrel = num.getSuperNo(theRegionLevel);
+
+  bool barrel_muon = (barrel == 1);
+  int maxLevel;
+  if (barrel_muon) {
+    maxLevel = theBChamberLevel;
+  } else {
+    maxLevel = theERollLevel;
+  }
+  if (num.getLevels() != maxLevel) {
+    abort();
```

Comment:
@slomeo @ianna why not throwing a cms::Exception?

Reply:
@fabiocos it was a  line code I used during debug...I'm sorry.. do I have to change it (and commit again)?

Reply:
@fabiocos fixed